docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1081 @@
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+Use the `term` suggester to suggest corrected spellings for individual words.
+The `term` suggester uses an edit distance to compute suggestions. Edit distance is the number of characters that need to be changed for a term to match.
+
+If a user misspells a search term:
+
+```json
+GET shakespeare/_search
+{
+  "suggest": {
+    "spell-check": {
+      "text": "lief",
+      "term": {
+        "field": "text_entry"
+      }
+    }
+  }
+}
+```
+
+Use the `term` suggester to return a list of corrections:
+
+```json
+{
+  "took": 48,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 0,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "suggest": {
+    "spell-check": [
+      {
+        "text": "lief",
+        "offset": 0,
+        "length": 4,
+        "options": [
+          {
+            "text": "lifes",
+            "score": 0.8,
+            "freq": 21
+          },
+          {
+            "text": "life",
+            "score": 0.75,
+            "freq": 805
+          },
+          {
+            "text": "lives",
+            "score": 0.6,
+            "freq": 187
+          },
+          {
+            "text": "liege",
+            "score": 0.6,
+            "freq": 138
+          },
+          {
+            "text": "lived",
+            "score": 0.6,
+            "freq": 80
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+The higher the score, the better the suggestion is. The frequency represents the number of times the term appears in the documents of that index.
+
+To implement a "Did you mean `suggestion`?" feature, use a `phrase` suggester.
+The `phrase` suggester is similar to the `term` suggester, except that it uses N-gram language models to suggest whole phrases instead of individual words.
+
+Create a custom analyzer called `trigram` that uses a `shingle` filter. This filter is similar to the `edge_ngram` filter, but it applies to words instead of letters:
+
+```json
+PUT shakespeare
+{
+  "settings": {
+    "index": {
+      "analysis": {
+        "analyzer": {
+          "trigram": {
+            "type": "custom",
+            "tokenizer": "standard",
+            "filter": [
+              "lowercase",
+              "shingle"
+            ]
+          }
+        },
+        "filter": {
+          "shingle": {
+            "type": "shingle",
+            "min_shingle_size": 2,
+            "max_shingle_size": 3
+          }
+        }
+      }
+    }
+  },
+  "mappings": {
+    "properties": {
+      "text_entry": {
+        "type": "text",
+        "fields": {
+          "trigram": {
+            "type": "text",
+            "analyzer": "trigram"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you run a query with an incorrect phrase:
```

Comment:
This example includes as incorrect phrase:

Reply:
Done